protocols/raft/src/test/java/io/atomix/protocols/raft/zeebe/util/ZeebeTestNode.java

```diff
@@ -0,0 +1,165 @@
+/*
+ * Copyright 2016-present Open Networking Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.atomix.protocols.raft.zeebe.util;
+
+import io.atomix.cluster.AtomixCluster;
+import io.atomix.cluster.ClusterMembershipService;
+import io.atomix.cluster.Member;
+import io.atomix.cluster.MemberId;
+import io.atomix.cluster.Node;
+import io.atomix.cluster.discovery.BootstrapDiscoveryProvider;
+import io.atomix.cluster.discovery.NodeDiscoveryProvider;
+import io.atomix.cluster.messaging.ClusterCommunicationService;
+import io.atomix.primitive.impl.ClasspathScanningPrimitiveTypeRegistry;
+import io.atomix.primitive.impl.DefaultPrimitiveTypeRegistry;
+import io.atomix.primitive.partition.ManagedPartitionGroup;
+import io.atomix.primitive.partition.ManagedPartitionService;
+import io.atomix.primitive.partition.impl.DefaultPartitionGroupTypeRegistry;
+import io.atomix.primitive.partition.impl.DefaultPartitionService;
+import io.atomix.protocols.raft.partition.RaftPartition;
+import io.atomix.protocols.raft.partition.RaftPartitionGroup;
+import io.atomix.protocols.raft.partition.impl.RaftPartitionServer;
+import io.atomix.protocols.raft.zeebe.partition.ZeebeRaftPartitionGroup;
+import io.atomix.storage.StorageLevel;
+import java.io.File;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.CompletableFuture;
+import java.util.stream.Collectors;
+
+public class ZeebeTestNode {
+
+  public static final String CLUSTER_ID = "zeebe";
+  private static final String DATA_PARTITION_GROUP_NAME = "data";
+  private static final String SYSTEM_PARTITION_GROUP_NAME = "system";
+  private static final String HOST = "localhost";
+  private static final int BASE_PORT = 10_000;
+  private final Member member;
+  private final Node node;
+  private final File directory;
+
+  private RaftPartitionGroup dataPartitionGroup;
+  private RaftPartitionGroup systemPartitionGroup;
+  private ManagedPartitionService partitionService;
+  private AtomixCluster cluster;
```

Comment:
Is it right that we have here an atomix cluster? Isn't this the wrapper around everything? Seems weird to have this in a protocol test :sweat_smile:

Reply:
Atomix is the wrapper around everything, no? And `AtomixCluster` is just the cluster management stuff? It was already a dependency of the module, since I guess it uses it for membership and communication. Or I misunderstood your question?

Reply:
Yes was just wondering since these protocol tests its like an interface knows its implementation class :sweat_smile: